### PR TITLE
#11247 inProgress scope is lost between different loaders.

### PR DESCRIPTION
**Pull request by jandvik**
Adding non-destructive onload and onerror callbacks

**What kind of change does this PR introduce?**
Changed LoadScriptRuntimeModule onload and onerror to non-destructively bind the onScriptComplete callback. With this change, existing callbacks will be retained in scope and executed in the order they were bound. This also ensures that all the "done" functions get called from the in-scope "inProgress".

**Did you add tests for your changes?**
No

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

**Comment by evilebottnawi** [comment]
What is a problem?

**Comment by jandvik** [comment]
@evilebottnawi - not sure what the problem is, I'm new to webpack contributions. Should I not be working off the master?

**Comment by jandvik** [comment]
looks like the Travis CI build does not like my reference to window for the global scope ... I should have probably looked a little deeper and used RuntimeGlobals?

**Comment by jandvik** [comment]
@sokra  @evilebottnawi at this point I'm fumbling around to try to fix the test errors. Spent more time trying to get that working than I did identifying the root cause of the bug and purposing a fix.

Some guidance would be extremely helpful at this point.

File: lib/runtime/LoadScriptRuntimeModule.js [hunk 1]
```diff
@@ -91,7 +91,13 @@ class LoadScriptRuntimeModule extends HelperRuntimeModule {
 		]);
 
 		return Template.asString([
-			"var inProgress = {};",
+			`var inProgress = {}`,
+			`if (${RuntimeGlobals.global}) {`,
+			Template.indent([
+				`inProgress = ${RuntimeGlobals.global}.__WEBPACK_LOADER_PROGRESS__ =`,
```
Comment (sokra) [inline lib/runtime/LoadScriptRuntimeModule.js:97]: Using a global is not acceptable. webpack should avoid globals. This only causes multiple webpack builds on the same page to conflict. Maybe not now, but in future when this mechnism is changed or whatever.

The script tag is the only "global" you can use. And use also spec-conform things on it.

> `script.onerror = script.onload = onScriptComplete;`

That the offending line. Find a better way to do that when others have always set `onerror/load`.

Store the old value for overriding and call it in the handler.

or use `script.addEventListener`, but here there are a few questions about:
* browser support?
* `// avoid mem leaks in IE.` `script.onerror = script.onload = null;` what is the reason behind that and does using `addEventListener` break it?

**Comment by jandvik** [comment]
@sokra  - thank you for your feedback. Between that and some sleep I found an approach that avoids global, while also being non-destructive to callbacks. :) 

I updated the test snapshots and was able to run the various "test:*" scripts locally and passing. Travis and Appveyor would not run locally for me though. Not sure what I'm missing and why the checks are not passing here now...

That being said, if you are satisfied with the approach I took in LoadScriptRuntimeModule.js, I can close this PR and create a fresh branch with all the changes in one commit. I also don't like seeing "17 commits" in one PR...

On another note, I did sign the CLA - not sure why that is still saying its not signed.

Hoping I can contribute more smoothly in the future when I come across issues.

**Comment by evilebottnawi** [comment]
@jandvik Please do not mix your work with other commits :smile:

**Comment by andvik** [comment]
@evilebottnawi how else do you resolve conflicts beside a rebase? Please enlighten me. This is also why I offered to close this PR and move my changes to a fresh branch off webpack's latest to avoid conflicts and rebases. I couldn't exactly find a list of things to do in the process, more being told "dont do that" when I try to figure things out myself. Perhaps there is a more in-depth contribution guide I can read than the one in the repo root folder?

**Comment by evilebottnawi** [comment]
just do rebase, not merge :smile:

**Comment by andvik** [comment]
It forced me to do both 🤷‍♂️

**Comment by andvik** [comment]
Would you tell me how to get te rest of the tests running locally so I can resolve the rest of the build issues, I'll move my changes to a fresh checkout and do it all in one clean commit so I can move on to my other work?

**Pull request closed without merging**